### PR TITLE
tfjob(v1alpha2): Add validation

### PR DESCRIPTION
**Pull request by gaocegege**
/assign @jlewi 

related to: kubeflow/tf-operator#437

Signed-off-by: Ce Gao <gaoce@caicloud.io>

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/874)

### CODE REVIEW

File: kubeflow/core/tf-job-operator.libsonnet [hunk 1]
```diff
@@ -55,6 +55,45 @@
...
+            properties: {
+              spec: {
+                properties: {
+                  tfReplicaSpecs: {
+                    properties: {
+                      Worker: {
+                        properties: {
+                          replicas: {
+                            type: "integer",
+                            minimum: 1,
```
Comment (jlewi) [inline kubeflow/core/tf-job-operator.libsonnet]: Please add comment explaining why we set minimum to 1.
Also please add a comment explaining why the validation doesn't extend to the pod template spec.

Comment (gaocegege) [reply kubeflow/core/tf-job-operator.libsonnet]: OK, I will update it.

**Comment by gaocegege** [comment]
PTAL

**Comment by gaocegege** [comment]
I do not think the failure is caused by this PR :thinking:

**Comment by jlewi** [comment]
Yeah test failures look like flakes building the bootstrapper image.

**Comment by gaocegege** [comment]
I am not sure when it will be fixed, could I retest the PR now?

**Comment by jlewi** [comment]
@gaocegege The failure is a networking issue downloading kubectl. So in principal retrying immediately should work.

**Comment by gaocegege** [comment]
I will rebase the master and have a try again.

**Comment by ankushagarwal** [comment]
https://github.com/kubeflow/kubeflow/pull/898 will hopefully fix the flaky test

**Comment by jlewi** [comment]
@gaocegege Can you sync past #898 to see if that fixed the flakes?

**Comment by gaocegege** [comment]
@jlewi OK, I will do it on Monday 😄

**Comment by gaocegege** [comment]
PTAL @jlewi 

The test is passed, thanks for your help @ankushagarwal

**Pull request merged**